src/store/models/lightning.ts

```diff
@@ -290,6 +293,108 @@ const lightningModel: LightningModel = {
       );
     },
   ),
+  addListeners: thunk(async (actions, network, { injections, getStoreState }) => {
+    const { nodes } = getStoreState().network.networkById(network.id);
+    nodes.lightning.forEach(
+      async node =>
+        await injections.lightningFactory.getService(node).addListenerToNode(node),
+    );
```

Comment:
Calling `await` in a `forEach` won't actually wait for the promises to resolve. Is this the behavior you intended? Right now, it is going to add the listeners while simultaneously subscribing to channel events. So it'll be a race to get there first and may produce unespected results. To actually make the execution wait for the promises to resolve, you can either use a `for/of` loop or you can use `await Promise.all(network.lightning.map(...))`

Reply:
True, no the behaviour I want. Updated to using a for/of